### DISCUSSION TITLE
Allow using templates for Header icons

### DISCUSSION THREAD

**Discussion by Mariusthvdb** [home-assistant/frontend#7569] [Other feature requests] [answered]
Please consider allowing templatable icons in the Header. 
Have been using Custom Header from the beginning, so am very much used to the great addition in functionality of changing icons in the Header. This allows for immediate feedback of certain states, and core functionality.

Fixed icons are so limited in current Home feedback, so this would be a very useful enhancement.

Now some of (now deprecated) CH's functionality is incorporated into Core HA, this would be a great moment to also embrace templatable icons.

some examples:
```
tab_icons:
  alarm: >-
    {{states('sensor.alarm_panel_icon')}}
  phones_tablets: >-
    {{states('sensor.presence_icon')}}
  weer_klimaat: >-
    {{states('sensor.weather_icon')}}
  alarmclock: >-
    {{'mdi:alarm-off' if is_state('sensor.next_alarm','Not set') else 'mdi:alarm'}}
```

resulting in:

```
    alarm_panel_icon:
      value_template: >
        {% set panel = states('alarm_control_panel.ha_rpi4_alarm') %}
        {% set mapper = {'disarmed':'mdi:shield-check',
                         'armed_away':'mdi:shield-lock',
                         'armed_custom_bypass':'mdi:security',
                         'armed_home':'mdi:shield-home',
                         'armed_night':'mdi:shield-home',
                         'arming':'mdi:shield-outline',
                         'triggered':'mdi:bell-alert'} %}
        {{mapper[panel] if panel in mapper else 'mdi:shield-check'}}

    presence_icon:
      friendly_name: Presence icon
      value_template: &icon
        >
          {% set presence = states('sensor.family_home') %}
          {% set mapper = {'0':'mdi:account-off',
                           '1':'mdi:account',
                           '2':'mdi:account-multiple',
                           '3':'mdi:account-multiple-check'} %}
          {{mapper[presence] if presence in mapper else 'mdi:account-group'}}

      weather_icon:
        value_template: >
          {% set mapper_icon =
            {'partly-cloudy-night':'night-partly-cloudy'} %}
          {% set mapper_br =
            {'pouring':'pouring',
             'lightning-rainy':'lightning-rainy',
             'snowy-rainy':'snowy-rainy'} %}
          {% set mapper_ds =
            {'clear-night':'night',
             'partlycloudy':'partly-cloudy'} %}
          {% set icon = states('sensor.dark_sky_icon') %}
          {% set buienradar = states('weather.buienradar') %}
          {% set dark = states('weather.dark_sky') %}

          {% set weather = mapper_icon[icon] if icon in mapper_icon else
                           mapper_br[buienradar] if buienradar in mapper_br else
                           mapper_ds[dark] if dark in mapper_ds else
                           dark if dark in
                             ['sunny','rainy','snowy','snowy-rainy','windy','fog','cloudy','hail','lightning']
                           else 'sunny-alert' %}
            mdi:weather-{{weather}}

      hour_icon:
        friendly_name: Hour icon
        value_template: &icon
          >
            {% set word = ['skipthis','one','two','three','four','five','six','seven',
                           'eight','nine','ten','eleven','twelve','one'] %}
            {%- set hour = now().strftime('%-I')|int %}
            {%- set minute = now().strftime('%-M')|int %}
            {%- set index = hour if minute <= 30 else hour + 1 %}mdi:clock-time-{{word[index]}}
              {{- '-outline' if states('sun.sun') != 'above_horizon' }}
        icon_template: *icon
```

One glance at the frontend immediately tell me how many of my family are home, what weather we have, and if my home is armed or not. Even what time it is...
To name but a few.

(btw, I also use colorings based on these same states, indicating severity, but using Custom-ui for that, so might not be possible in core without? Or..?)

thanks for considering

<img width="514" alt="Schermafbeelding 2020-11-01 om 11 25 54" src="https://user-images.githubusercontent.com/33354141/97800379-1b316b00-1c35-11eb-9fbf-a663a4d8bfb2.png">

<img width="541" alt="Schermafbeelding 2020-11-01 om 11 28 42" src="https://user-images.githubusercontent.com/33354141/97800435-78c5b780-1c35-11eb-8718-351f3d38605a.png">

**Comment by iantrich** [comment, marked as answer]
This is far too advanced/complex for core. I see no reason that someone could not extract the related logic from custom header into a custom plug-in to do this which is a much better suited solution.

**Reply by Mariusthvdb** [reply]
Thanks. Still, as a throwback to earlier days: we could do this in States , when the icon was a group icon, and could be simply customized.

Why then would a template for the View icon be too complicated now?

**Reply by frenck** [reply]
Because in the early days the backend made the frontend, now it is the other way around.
As said, this is too complex for our core feature set.

**Reply by Mariusthvdb** [reply]
not really sure what you mean by 'now it is the other way around' (because that customization of the Group view also was in the Frontend before), but I get the effort is futile here ;-)

just as a general question on the matter: what is considered to be 'too complex for the core feature set' and what is not? Is this the outcome of some architecture decision? If so I'd be very interested to read up on that.

will have a look at Ian's challenge above to take bits and pieces of the Custom Header plug-in, though this is way beyond my skill set.
thanks.
